### PR TITLE
DEV: dry up the Do Not Disturb widget

### PR DESCRIPTION
**Pull request by AndrewPrigorshnev**

### CODE REVIEW

File: app/assets/javascripts/discourse/app/widgets/do-not-disturb.js [hunk 1]
```diff
@@ -11,21 +11,10 @@ export default createWidget("do-not-disturb", {
...
-        h("button.btn-flat.do-not-disturb-inner-container", [
-          iconNode("toggle-off"),
-          h("span.do-not-disturb-label", I18n.t("pause_notifications.label")),
-        ]),
-      ];
-    }
+    const isOn = this.currentUser.isInDoNotDisturb();
+    const button = this._menuButton(isOn);
+
+    return [button];
```
Comment (nbianca) [inline app/assets/javascripts/discourse/app/widgets/do-not-disturb.js:17]: Just a personal preference, but this function looks longer than it should be.

```suggestion
    return [this._menuButton(isOn)];
```

File: app/assets/javascripts/discourse/app/widgets/do-not-disturb.js [hunk 2]
```diff
@@ -11,21 +11,10 @@ export default createWidget("do-not-disturb", {
...
-        h("button.btn-flat.do-not-disturb-inner-container", [
-          iconNode("toggle-on"),
-          this.label(),
-        ]),
-      ];
-    } else {
-      return [
-        h("button.btn-flat.do-not-disturb-inner-container", [
-          iconNode("toggle-off"),
-          h("span.do-not-disturb-label", I18n.t("pause_notifications.label")),
```
Comment (nbianca) [inline app/assets/javascripts/discourse/app/widgets/do-not-disturb.js:25]: I do not see this string used anymore if the "do not disturb" mode is not enabled. Can you confirm it is still in use?

**Review by nbianca — APPROVED** [review]

**Comment by jjaffeux** [comment]
@AndrewPrigorshnev I think we should try to merge this Monday

File: app/assets/javascripts/discourse/app/widgets/do-not-disturb.js (hunk 2 above)
Comment (AndrewPrigorshnev) [reply app/assets/javascripts/discourse/app/widgets/do-not-disturb.js:25]: We use it in a couple of places: below in the `_label` function and also in the `profile-tab-content` component.

**Comment by AndrewPrigorshnev** [comment]
Test failures are unrelated, we're having them on other PRs as well currently. @jjaffeux or @nbianca could you please approve it, I've addressed review comments.

**Comment by AndrewPrigorshnev** [comment]
Oh, this actually was already approved, forgot about it.

**Pull request merged**